pilot/pkg/serviceregistry/kube/controller/multicluster.go

```diff
@@ -203,3 +254,76 @@ func (m *Multicluster) GetRemoteKubeClient(clusterID string) kubernetes.Interfac
 	}
 	return nil
 }
+
+func (m *Multicluster) createRemoteConfigController(clusterID string, config *clientcmdapi.Config, options Options) (model.ConfigStoreCache, error) {
+	schemas := collection.NewSchemasBuilder()
+	if features.EnableServiceApis {
+		schemas = schemas.
+			MustAdd(collections.K8SServiceApisV1Alpha1Tcproutes).
+			MustAdd(collections.K8SServiceApisV1Alpha1Gatewayclasses).
+			MustAdd(collections.K8SServiceApisV1Alpha1Gateways).
+			MustAdd(collections.K8SServiceApisV1Alpha1Httproutes).
+			MustAdd(collections.K8SServiceApisV1Alpha1Trafficsplits)
+	}
+	for _, schema := range collections.Pilot.All() {
+		if err := schemas.Add(schema); err != nil {
+			return nil, err
+		}
+	}
+	remoteConfig := clientcmd.NewDefaultClientConfig(*config, &clientcmd.ConfigOverrides{})
+	restConfig, err := remoteConfig.ClientConfig()
+	if err != nil {
+		return nil, fmt.Errorf("kubeconfig for remote cluster %v is not valid: %v", clusterID, err)
```

Comment:
What user can do to fix this error when the error occurs?

Reply:
will add message to let user check their remote secret.